File: client/angular-dart/qmeAngularDart/lib/component/qme_register.dart
part of qme;

@Component(
    selector: 'qme-register',
    templateUrl: 'packages/qme/component/qme_register.html',
    useShadowDom: false)
class QMeRegisterComponent {

    QMeUser user = new QMeUser();

    NgForm registerForm;

    //Router router;

    //QMeRegisterComponent(Router router) {
    //    this.router = router;
    //}

    void registerUser() {
      if (!QmeErrorHolder.instance.isError) {
        print(user.userEmail);
        print(user.userPassword);
        QmeErrorHolder.instance.setError("some error message");
      }
    }

    void validatePassword() {
        if(user.userPassword != user.userPasswordConfirm){
          QmeErrorHolder.instance.setError("Password do not match, please confirm password");
          user.userPasswordConfirm ="";
        }else{
          QmeErrorHolder.instance.removeError();
        }

    }


}

File: client/angular/qmeAngular/app/js/components/register/qmeregister.js
(function () {
    'use strict';

    qmeApp.controller('qmeRegister', QMeRegisterController);

    QMeRegisterController.$inject = ['qmeFlashService'];

    function QMeRegisterController(qmeFlashService) {

        var qmeRegister = this;

        qmeRegister.userEmail = "";
        qmeRegister.userName = "";
        qmeRegister.userPassword = "";
        qmeRegister.userPasswordConfirm = "";
        qmeRegister.userFirstName = "";
        qmeRegister.userLastName = "";

        qmeRegister.validatePassword = function (){
            if(qmeRegister.userPassword != qmeRegister.userPasswordConfirm){
                qmeFlashService.Error("Password do not match, please confirm password");
                qmeRegister.userPasswordConfirm ="";
            }else{
                qmeFlashService.Clear();
            }
        }

        qmeRegister.registerUser = function (){
            //TODO:
            //Make http call to service and log in user
            //qmeFlashService.Error("testing");
            //qmeHeader.userName = "tocallservice";
        }

    }

})();